app/cosmo/base.toml

```diff
@@ -0,0 +1,1383 @@
+target = "thumbv7em-none-eabihf"
+chip = "../../chips/stm32h7"
+memory = "memory-large.toml"
+stacksize = 896
+fwid = true
+
+[kernel]
+name = "cosmo"
+requires = {flash = 32768, ram = 8192}
+features = ["dump"]
+
+[mmio]
+peripheral-region = "fmc_nor_psram_bank_1"
+register-map = "../../drv/spartan7-loader/cosmo-seq/cosmo_seq_top.json"
+
+[caboose]
+tasks = ["control_plane_agent"]
+region = "flash"
+size = 256
+default = true
+
+[tasks.jefe]
+name = "task-jefe"
+priority = 0
+max-sizes = {flash = 16384, ram = 2048}
+start = true
+features = ["dump"]
+stacksize = 1536
+notifications = ["fault", "timer"]
+extern-regions = ["sram1", "sram2", "sram3", "sram4"]
+
+[tasks.jefe.config.on-state-change]
+net = "jefe-state-change"
+host_sp_comms = "jefe-state-change"
+
+[tasks.jefe.config.allowed-callers]
+set_state = ["cosmo_seq"]
+set_reset_reason = ["sys"]
+request_reset = ["hiffy", "control_plane_agent"]
+
+[tasks.net]
+name = "task-net"
+stacksize = 8000
+priority = 5
+features = ["mgmt", "h753", "cosmo", "vlan", "vpd-mac"]
+max-sizes = {flash = 131072, ram = 65536, sram1_mac = 16384}
+sections = {eth_bulk = "sram1_mac"}
+uses = ["eth", "tim16"]
+start = true
+interrupts = {"eth.irq" = "eth-irq", "tim16.irq" = "mdio-timer-irq"}
+task-slots = ["sys", "packrat", { spi_driver = "spi2_driver" }, "jefe"]
+notifications = ["eth-irq", "mdio-timer-irq", "wake-timer", "jefe-state-change"]
+
+[tasks.sys]
+name = "drv-stm32xx-sys"
+features = ["h753", "exti", "no-panic"]
+priority = 1
+uses = ["rcc", "gpios", "system_flash", "syscfg", "exti"]
+start = true
+task-slots = ["jefe"]
+notifications = ["exti-wildcard-irq"]
+
+[tasks.sys.interrupts]
+"exti.exti0" = "exti-wildcard-irq"
+"exti.exti1" = "exti-wildcard-irq"
+"exti.exti2" = "exti-wildcard-irq"
+"exti.exti3" = "exti-wildcard-irq"
+"exti.exti4" = "exti-wildcard-irq"
+"exti.exti9_5" = "exti-wildcard-irq"
+"exti.exti15_10" = "exti-wildcard-irq"
+
+[tasks.sys.config.gpio-irqs.rot_irq]
+port = "F"
+pin = 2
+owner = {name = "sprot", notification = "rot_irq"}
+
+[tasks.spi2_driver]
+name = "drv-stm32h7-spi-server"
+priority = 3
+max-sizes = {flash = 16384, ram = 4096}
+features = ["spi2", "h753"]
+uses = ["spi2"]
+start = true
+interrupts = {"spi2.irq" = "spi-irq"}
+stacksize = 872
+task-slots = ["sys"]
+notifications = ["spi-irq"]
+
+# XXX this is only used by cosmo_seq; could we merge it?
```

Comment:
XXX comment

Reply:
This remains a potential future improvement, but I don't want to handle it right now!